packages/app/features/dashboard/profile/textEllipsis.tsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from "react";
+import { insert, join, pipe, remove, split } from "ramda";
+import { Text, View } from "app/design-system";
+import { tw } from "app/design-system/tailwind";
+
+export function TextEllipsis({ text }: { text: string }) {
+  const [wrapperWidth, setWrapperWidth] = useState(0);
+  const [displayedText, setDisplayedText] = useState("A");
+  const [charLength, setCharLength] = useState(0);
+
+  useEffect(() => {
+    if (!charLength || !wrapperWidth) return;
+    const availableTextLength = wrapperWidth / charLength;
+    if (availableTextLength >= text.length) {
+      setDisplayedText(text);
+      return;
+    }
+
+    const lengthToCut = text.length - availableTextLength + 2;
```

Comment:
why do we add `2` to the result?

Reply:
`.` is a bit narrower  so i put 2 instead of 3

Reply:
can we put it in const so it will be more descriptive? Like `TextEllipsisLength` or something